presto-main/src/main/java/com/facebook/presto/type/DecimalParametricType.java

```diff
@@ -37,26 +36,21 @@ public String getName()
     }
 
     @Override
+    @NotNull
     public Type createType(List<TypeParameter> parameters)
     {
+        checkArgument(parameters.size() <= 2, "Expected no more than 2 literal parameters for DECIMAL");
```

Comment:
replace `if/else` sequence with `switch(parameters.size())`; and put `throw IllegalArguementException` in `default` section.